### ISSUE TITLE
[Tabs] MDCTabBar should support dynamic type for its items

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#3041]
### Overview
The MDCTabBar does not support dynamic type which can make it difficult for users with visual impairments to read the text.  MDCTabBar should support a top-level property similar to `mdc_adjustsFontForContentSizeCategory` which permits automatic adjustment of font types for dynamic type.  The property should support UIAppearance as well.

### Reproduction steps
1. Configure the device to have the largest dynamic type size possible (in Settings App)
1. Open the Tabs "Demo" example
1. The Tabs should have larger text

### Version number
46.0.0

**Comment by ianegordon** [comment]
Apple's Navigation Elements do not typically support Dynamic Type.  We are following their pattern.

**Comment by romoore** [comment]
@ianegordon Can you please close this issue if you can confirm it's a won't-fix?

**Comment by jverkoey** [comment]
@ianegordon Please re-open if this is something we wish to pursue.
